src/AstTranslator.cpp

```diff
@@ -60,6 +60,10 @@ IODirectives getInputIODirectives(const AstRelation* rel, std::string filePath =
     const std::string inputFilePath = (filePath.empty()) ? Global::config().get("fact-dir") : filePath;
     const std::string inputFileExt = (fileExt.empty()) ? ".facts" : fileExt;
 
+    const bool isIntermediate =
+            (Global::config().has("engine") && inputFilePath == Global::config().get("output-dir") &&
```

Comment:
It is valid, and even reasonable, to have your input files in the same directory as the output and to name them with a ".facts" suffix. If you're going to use path as a test then you need something more unique - or at least more unreasonable for standard use.

Reply:
I agree, but this is a little more tricky that it first appears. 

Consider the following program, where Souffle is called with `-D./mydir F./mydir -efile`. 

~~~
.decl A(x:number)
.input A

.decl B(x:number)
.output B

A(1).
B(x) :- A(x).
~~~

Here, `A` becomes an intermediate relation, formed from both the input file and the fact `A(1)`. If `A` is written out for `B` to consume, it will override the `A.facts` file in `mydir`. Disallowing input and output directories to be the same if `-e` is used is the only way to ensure this does not occur, under the current way in which intermediate relations are read and written. It might be good to change the way in which this is done in future, however doing so now would require substantial changes to this PR, and would take some time.